Controllers/SearchForStaff.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using stunning_robot_HR.Data;
+using stunning_robot_HR.Models;
+
+
+namespace stunning_robot_HR.Controllers
+{
+    public class SearchForStaff : Controller
+    {
+        private stunning_robot_HRContext _context;
+        private IStaffRepository staffRepository;
+        
+        public SearchForStaff(stunning_robot_HRContext context)
+        {
+            _context = context;
+            staffRepository = new StaffRepository(_context);
+        }
+
+        public async Task<IActionResult> GetStaffByFullName(string searchString)
```

Comment:
I believe this is the method we refactored together. But having the repo return a collection is definitely the right call here.